plugins/woocommerce/src/Blocks/BlockTypes/ProductFilters.php

```diff
@@ -83,20 +115,63 @@ protected function render( $attributes, $content, $block ) {
 			return $content;
 		}
 
+		if ( $this->collection_data_is_empty( $attributes ) ) {
+			return $this->render_empty_block();
+		}
+
+		return $this->render_filter_block( $content, $block );
+	}
+
+	/**
+	 * Reset the current response, must be done before rendering.
+	 *
+	 * @return void
+	 */
+	private function reset_current_response() {
 		/**
-		 * At this point, WP starts rendering the Collection Filters block,
+		 * When WP starts rendering the Product Filters block,
 		 * we can safely unset the current response.
 		 */
 		$this->current_response = null;
+	}
+
+	/**
+	 * Render the block when it's empty.
+	 *
+	 * @return string - Rendered block type output.
+	 */
+	private function render_empty_block() {
+		$this->reset_current_response();
+
+		return sprintf(
+			'<nav %1$s></nav>',
+			get_block_wrapper_attributes(
+				array(
+					'data-wc-interactive' => wp_json_encode( array( 'namespace' => $this->get_full_block_name() ) ),
+					'class'               => 'wc-block-product-filters',
```

Comment:
Empty block is missing navigation id. Looks like it causes the following issue:

https://github.com/woocommerce/woocommerce/assets/5423135/d7daf8e5-7bfe-43ea-8ad6-ae15a21e4a4b

Reply:
Resolved [here](https://github.com/woocommerce/woocommerce/pull/43688/commits/8a6aa4fbea3881bf630e67cf396a6a0da5fa56c6)